src/maintainer/knowledge_base.rst

```diff
@@ -791,6 +793,52 @@ Following implies that ``python`` is a runtime dependency and a Python matrix fo
 
 You need to rerender the feedstock after this change.
 
+Using CMake
+===========
+
+`CMake <https://cmake.org/>`__ can be used to build more complex projects in ``build.sh``
+or ``bld.bat`` scripts.
+
+If you are using cmake, be sure to make it a build requirement in the ``build`` or ``host`` section.
+
+.. code-block:: yaml
+
+    host:
+      - cmake
+
+CMake requires you to supply a configuration file, usually called ``CMakeLists.txt``.
+You can use the one from your source tar ball if it exists or include your own in the
+same directory as the conda-forge recipe file and copy it to the source directory.
```

Comment:
I don't think we need to encourage writing CMake configuration files for packages. Instead we should recommend that packages should use CMake if the upstream build system is CMake or if the upstream doesn't have a build system.

Reply:
Again, I'm ok changing this, and I'll go ahead and do that. But... there are a few reasons this might be useful "as is" or with  a slightly modified comment. Sometimes there is already a package release with a tar ball that has no CMake support or CMake that is not compatible with conda-forge. In this case, it's useful to know that you can include the config file near the recipe and use it, particularly when debugging a recipe. (I've been thinking of writing a 'debugging recipes' section too btw).

Reply:
I'm torn. There are people who did write their own `CMakeLists.txt` to get things to build in conda-forge in the early days. However I agree with Isuru, this is scope creep and it seems hard to stay on top of. Maybe we can link out to some CMake docs somewhere?

Reply:
I guess one question would be "who is the audience?" If it's msbuild users switching to cmake, then I think it should stay; but it sounds like the msbuild problem might be resolved soon anyway. So then it's more likely that the audience will be people adding projects with existing cmake builds.

Reply:
I removed this section too.

Reply:
I think that is the case (even outside of the msbuild problem).

In any event, we probably will wind up solving the msbuild problem by trying to migrate things to CMake if they already support CMake as opposed to adding that support for projects. Cases without CMake as an option will likely get solved in a different way (likely moving to vs2017).